Project Finished

Jackua opened the pull request:
Completed the following challenges

- Allow users to create their own habits to track
- Seed data into database

Thank you for the review!

chrisjamiecarter commented on HabitTracker.Jackua/Controller/HabitTrackerController.cs:
```diff
@@ -0,0 +1,542 @@
+
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
```
This code is open to [SQL Injection Attacks](https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html)

Remember to parameterise your queries:
```
// Open to SQL Injection Attacks.
command.CommandText = $"INSERT INTO Table(Column) VALUES ('{value}');";

// vs

// Closed to SQL Injection Attacks.
command.CommandText = $"INSERT INTO Table(Column) VALUES (@Value);";
command.Parameters.AddWithValue("@Value", value);
```

chrisjamiecarter commented on HabitTracker.Jackua/View/MenuView.cs:
```diff
@@ -0,0 +1,113 @@
+            Console.WriteLine("Type 9 to View Summary of Records");
+            DashLines();
+        }
+        public static void InvalidCommand()
```
Remember to have one empty line between methods.

chrisjamiecarter reviewed (changes requested):
Hey @Jackua,

I have added some minor comments to help support your growth as a developer.

For me, consistent code style is really important, and you have done a great job with this application. Well done.

**Bugs**
Unfortunately, I did spot a bug within your application:

When I try update a record I get the below error:
![image](https://github.com/user-attachments/assets/7ace1f59-74a6-4ecb-a2a8-89c729932860)

Could you please revisit this to try understand and fix any issues, as this currently fails two requirements:

- [ ] The users should be able to insert, delete, update and view their logged habit.
- [ ] You should handle all possible errors so that the application never crashes.

Feel free to reach out if you have any questions or if you'd like to collaborate on any of these points.

Thanks
@chrisjamiecarter

chrisjamiecarter commented:
@Jackua, 

Some point on this apps User Experience:

- I would clear the console each time before you write the main menu. This allows the user to segment an action, maintain readability of the console window and improve the experience for the user.
- Look into cascading deletes. So that if I wanted to delete a habit in the database that had records. The app would prompt me to confirm if I was sure I really wanted to do this (as it is a destructive action), and if I choose to, it would delete the habit, and all associated records.
- When showing a report/table like in View All Records, View All Habits and View Summary of Records, put a `Console.WriteLine("Press any key to continue...");` and a `Console.Read();`. This allows the user to digest the information they requested without having the main menu smack them in the face.

Again, these are only suggestions, to support you growth as a developer.

Thanks,
@chrisjamiecarter

Jackua commented:
Hi Chris,

Thank you for the feedback!

I added a README, fixed the bug, cleaned up indentation levels, parameterize all the queries, and made updates based on your suggestions.

chrisjamiecarter reviewed (approved):
Hey @Jackua 👋 ,

Well done on the updates.

This resolves the bug and fulfills all the requirements.

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter
